akka-http-core/src/main/scala/akka/http/scaladsl/model/Multipart.scala

```diff
@@ -187,10 +187,7 @@ object Multipart {
 
   private def strictify[BP <: Multipart.BodyPart, BPS <: Multipart.BodyPart.Strict](parts: Source[BP, Any])(f: BP ⇒ Future[BPS])(implicit fm: Materializer): Future[Vector[BPS]] = {
     import fm.executionContext
-    // TODO: move to Vector `:+` when https://issues.scala-lang.org/browse/SI-8930 is fixed
-    parts.runFold(new VectorBuilder[Future[BPS]]) {
-      case (builder, part) ⇒ builder += f(part)
-    }.fast.flatMap(builder ⇒ FastFuture.sequence(builder.result()))
+    parts.mapAsync(Int.MaxValue)(f).runWith(Sink.seq).fast.map(_.toVector)
```

Comment:
Hmm, I think I got that wrong for byte ranges since we make sure that they don’t overlap. But this code is not only for those, and a general `strictify` method must assume the streams to be coupled.